src/Microsoft.AspNetCore.WebHooks.Receivers/WebHookConstants.cs

```diff
@@ -32,43 +34,61 @@ public static class WebHookConstants
-        /// <para>
-        /// Key corresponds to an <see cref="Extensions.Configuration.IConfiguration"/> entry in the default
-        /// <see cref="IWebHookReceiverConfig"/> implementation
-        /// </para>
         /// </summary>
         /// <remarks>
         /// Most, if not all, receiver configurations include
         /// <see cref="Filters.WebHookSecurityFilter.EnsureSecureConnection"/> calls. For example,
         /// <see cref="Filters.WebHookVerifyCodeFilter"/> calls the method if
         /// <see cref="Metadata.IWebHookSecurityMetadata.VerifyCodeParameter"/> is <see langword="true"/>.
         /// </remarks>
-        public static string DisableHttpsCheckConfigurationKey => "MS_WebHookDisableHttpsCheck";
+        public static string DisableHttpsCheckConfigurationKey { get; } = ConfigurationPath.Combine(
+            ReceiverConfigurationSectionKey,
+            "DisableHttpsCheck");
 
         /// <summary>
-        /// Gets the prefix of keys used to retrieve a per-receiver or per-id configuration string. Configuration value
-        /// often contains the receiver's secret key or keys. Key corresponds to an
-        /// <see cref="Extensions.Configuration.IConfiguration"/> entry in the default
-        /// <see cref="IWebHookReceiverConfig"/> implementation.
+        /// Gets the key of the <see cref="IConfigurationSection"/> containing all configuration values used in this
+        /// package. Immediate children include global configuration values and receiver-specific
+        /// <see cref="IConfigurationSection"/>s.
         /// </summary>
-        public static string ReceiverConfigurationKeyPrefix => "MS_WebHookReceiverSecret_";
+        public static string ReceiverConfigurationSectionKey => "WebHooks";
+
+        /// <summary>
+        /// Gets the relative key name of the <see cref="IConfigurationSection"/> containing secret keys. This section
+        /// is an immediate child of the receiver-specific <see cref="IConfigurationSection"/>. For most receivers,
+        /// immediate children of this section are id and secret key pairs.
+        /// </summary>
+        public static string SecretKeyConfigurationKeySectionKey => "SecretKey";
 
         /// <summary>
         /// Gets the name of the <see cref="AspNetCore.Routing.RouteValueDictionary"/> entry containing the event name
-        /// for the current request.
+        /// for the current request when there is a single event name.
         /// </summary>
         /// <seealso cref="Metadata.IWebHookEventMetadata"/>
         public static string EventKeyName => "event";
 
         /// <summary>
+        /// Gets the names of the <see cref="AspNetCore.Routing.RouteValueDictionary"/> entries containing event names
+        /// for the current request when there are more than one event name.
+        /// </summary>
+        /// <seealso cref="Metadata.IWebHookEventMetadata"/>
+        public static string[] EventKeyNames { get; } = Enumerable.Range(0, 100)
```

Comment:
>EventKeyNames [](start = 31, length = 13)

Why do we need this as a public constant? why would someone developing a receiver care

Reply:
Because otherwise every request would create at least two `string`s per entry in the array. And requests an contain numerous notifications / event names.

Reply:
Also, users can write their own receivers to special-case event names in some way.

The alternative is to add a `SetEventNames(...)` extension method to `WebHookRouteDataExtensions`. Does that feel better to you?